### DISCUSSION TITLE
Doing data analysis

### DISCUSSION THREAD

**Discussion by mubarak-mustopha** [microsoft/ML-For-Beginners#436] [Q&A]
I always get stuck whenever I'm trying do some data analysis before actually building a model.
Like I don't really know the right questions to ask based on the data I have.....can anyone make a recommendation or a resource that can help me solve this problem?

**Comment by yuridesideri** [comment]
Try reflecting more on gap-analysis

**Reply by jlooper** [reply]
Maybe this lesson might be of help: https://github.com/microsoft/Data-Science-For-Beginners/tree/main/4-Data-Science-Lifecycle/15-analyzing

**Reply by mubarak-mustopha** [reply]
@jlooper thanks alot.

**Reply by mubarak-mustopha** [reply]
@yuridesideri sorry what do you mean by that?

**Reply by yuridesideri** [reply]
Google's definition: A gap analysis is the process companies use to compare their current performance with their desired, expected performance.

What I want to say is try reflecting on what are your goals and what obstacles there are in front of you. Then you can start creating questions on those obstacles.

**Reply by mubarak-mustopha** [reply]
@yuridesideri ooh got it... thanks for that buddy.
